instrumentation-api-annotation-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/MethodSpanAttributesExtractorBuilder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.annotation.support;
+
+import io.opentelemetry.instrumentation.api.caching.Cache;
+import java.lang.reflect.Method;
+
+/** A builder of {@link MethodSpanAttributesExtractor}. */
+public final class MethodSpanAttributesExtractorBuilder<REQUEST, RESPONSE> {
+  MethodExtractor<REQUEST> methodExtractor;
+  MethodArgumentsExtractor<REQUEST> methodArgumentsExtractor;
+  Cache<Method, AttributeBindings> cache;
+  ParameterAttributeNamesExtractor parameterAttributeNamesExtractor;
+
+  public MethodSpanAttributesExtractorBuilder(MethodExtractor<REQUEST> methodExtractor) {
+    this.methodExtractor = methodExtractor;
+  }
+
+  /** Sets the {@link Cache} to cache the {@link AttributeBindings} by {@link Method}. */
+  public MethodSpanAttributesExtractorBuilder<REQUEST, RESPONSE> setCache(
```

Comment:
Do we need to make this configurable?

Reply:
Probably not, at least at this point.  I was considering whether I might want to at least be able to disable caching for unit tests when I might eventually merge `BaseAttributeBinder` into this class.  I am not touching that at this point (as I don't want to affect `opentelemetry-annotations-1.0` yet) so I can remove this method and default the behavior to caching until I have a reason to do otherwise.

Reply:
I nixed for now, can revisit later if need be.